google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/MerchantCenterDynamicRemarketing.java

```diff
@@ -0,0 +1,384 @@
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.merchantCenterAccountId = Long.parseLong("INSERT_MERCHANT_CENTER_ID_HERE");
+      params.campaignBudgetId = Long.parseLong("INSERT_CAMPAIGN_BUDGET_ID_HERE");
+      params.userListId = Long.parseLong("INSERT_USER_LIST_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+        "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new MerchantCenterDynamicRemarketing().runExample(
+        googleAdsClient,
+        params.customerId,
+        params.merchantCenterAccountId,
+        params.campaignBudgetId,
+        params.userListId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+        "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+        gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(
+    GoogleAdsClient googleAdsClient,
+    long customerId,
+    long merchantCenterId,
+    long budgetId,
+    long userListId) throws IOException {
+      String campaignResourceName = createCampaign(
+        googleAdsClient, customerId, merchantCenterId, budgetId);
+      String adGroupResourceName = createAdGroup(googleAdsClient, customerId, campaignResourceName);
+      createAd(googleAdsClient, customerId, adGroupResourceName);
+      attachUserList(googleAdsClient, customerId, adGroupResourceName, userListId);
+  }
+
+  /**
+   * Creates a campaign linked to a Merchant Center product feed.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param merchantCenterId the merchant center account ID.
```

Comment:
* @param merchantCenterAccountId the Merchant Center account ID.

"Merchant Center" should be initial upper case as consistent with other code samples.